test/test_mp_all_reduce_xla_backend.py

```diff
@@ -0,0 +1,35 @@
+import os
+import sys
+import torch
+import torch_xla
+import torch_xla.core.xla_model as xm
+import torch_xla.distributed.xla_multiprocessing as xmp
+import torch_xla.distributed.xla_backend
+import torch.distributed as dist
+
+
+def _mp_fn(index):
+  device = xm.xla_device()
+  if xm.xla_device_hw(device) in ('TPU', 'GPU'):
+    world_size = xm.xrt_world_size()
+    rank = xm.get_ordinal()
+
+    print(f'RANK {rank} init\'ing torch.distributed...')
+    dist.init_process_group('xla', world_size=world_size, rank=rank)
+    print(f'RANK {rank} init\'ed torch.distributed.')
+
+    ones = torch.ones((2, 3))
+    print("transferring ones to device")
+    xones = ones.to(device)
+    print("running all reduce")
+    dist.all_reduce(xones)
```

Comment:
(can be done in followup PRs) do we want to test running multiple allreduce lazily in a loop and all of their outputs are expected?

Reply:
Sure. Added test_mp_multi_all_reduce_xla_backend.py